csharp/src/Ice/ConnectionI.cs

```diff
@@ -2085,8 +2094,44 @@ private int sendNextMessage(out Queue<OutgoingMessage> callbacks)
 
     private int sendMessage(OutgoingMessage message)
     {
+        Debug.Assert(_state >= StateActive);
         Debug.Assert(_state < StateClosed);
 
+        bool isHeartbeat = message.stream.getBuffer().b.get(8) == Protocol.validateConnectionMsg;
+        if (!isHeartbeat)
+        {
+            cancelInactivityTimer();
```

Comment:
Can we unschedule without disposing of the timer in this case?

Reply:
That's not the logic on the code here. Unschedule = dispose and set to null. Schedule = create a new timer.
And we rely on this logic in inactivityCheck when we compare inactivityTimer with _inacvityTimer.

The scheduling is very infrequent so there is no need to worry about reusing the timer object.